### PR TITLE
🚑 Forgot/Reset password fix for MongoDB - only update mandatory fields

### PR DESCRIPTION
**Pull request by damtsnkff**
#### Description of what you did:
On both the `/forgot-password` and `/reset-password` endpoints, a 500 error happened because of the way the user was updated.
The error had the following message.
```
Error: Argument passed in must be a single String of 12 bytes or a string of 24 hex characters
```

Updated: `changePassword` and `forgotPassword` methods in `user-permissions/controllers/Auth.js`.

Fix https://github.com/strapi/strapi/issues/5249 

#### Step to reproduce issue
- Fresh install of a Strapi project (with mongoDB config).
- Install the sengrid email provider (bc somehow with default sendmail provider doesn't work)
- Create a user POST /auth/local/register --> Ok
- Try forgot password POST /auth/forgot-password --> Ok
- Add a new collection for example Bookings with a field relation has and belong to one with the users collection.
- Try the forgot password /auth/forgot-password ---> Error

### CODE REVIEW

**Comment by damtsnkff** [comment]
The travis CI seems buggy or did I mess something up?
Happy to correct / fix it if needed.

File: packages/strapi-plugin-users-permissions/controllers/Auth.js [hunk 1]
```diff
@@ -363,7 +368,7 @@ module.exports = {
     }
 
     // Update the user.
-    await strapi.query('user', 'users-permissions').update({ id: user.id }, user);
+    await strapi.query('user', 'users-permissions').update({ id: user.id }, { resetPasswordToken });
```
Comment (alexandrebodin) [inline packages/strapi-plugin-users-permissions/controllers/Auth.js:371]: you can remove the user.resetPasswordToken : resetPasswordToken if you do that :)

File: packages/strapi-plugin-users-permissions/controllers/Auth.js [hunk 2]
```diff
@@ -206,7 +206,12 @@ module.exports = {
       });
 
       // Update the user.
-      await strapi.query('user', 'users-permissions').update({ id: user.id }, user);
+      await strapi
+        .query('user', 'users-permissions')
+        .update(
+          { id: user.id },
+          { resetPasswordToken: user.resetPasswordToken, password: user.password }
```
Comment (alexandrebodin) [inline packages/strapi-plugin-users-permissions/controllers/Auth.js:213]: You can avoid mutating the user her. and just used two variables for reset token and password

File: packages/strapi-plugin-users-permissions/controllers/Auth.js (hunk 1 above)
Comment (damtsnkff) [reply packages/strapi-plugin-users-permissions/controllers/Auth.js:371]: Same as above, I wasn't sure the property set on the `user` wasn't used line 344. (Not familiar with the lodash / the syntax). Fixing them now.

**Comment by damtsnkff** [comment]
@alexandrebodin thanks for the review.
I wasn't sure if the properties on the user were used or not, but I've now deleted the lines mutating the `user` object.

**Pull request merged**